src/test/java/edu/cornell/kfs/fp/document/CuDisbursementVoucherDocumentTest.java

```diff
@@ -0,0 +1,398 @@
+package edu.cornell.kfs.fp.document;
+
+import com.rsmart.kuali.kfs.fp.businessobject.DisbursementVoucherDocumentExtension;
+import edu.cornell.kfs.fp.businessobject.CuDisbursementPayee;
+import edu.cornell.kfs.fp.businessobject.CuDisbursementVoucherPayeeDetail;
+import edu.cornell.kfs.fp.businessobject.CuDisbursementVoucherPayeeDetailExtension;
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+import org.apache.commons.lang.StringUtils;
+import org.easymock.EasyMock;
+import org.easymock.IMockBuilder;
+import org.easymock.Mock;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.kuali.kfs.fp.businessobject.DisbursementPayee;
+import org.kuali.kfs.fp.businessobject.DisbursementVoucherPayeeDetail;
+import org.kuali.kfs.fp.businessobject.PaymentReasonCode;
+import org.kuali.kfs.fp.document.DisbursementVoucherDocument;
+import org.kuali.kfs.fp.document.service.DisbursementVoucherPayeeService;
+import org.kuali.kfs.fp.document.service.DisbursementVoucherPaymentReasonService;
+import org.kuali.kfs.sys.KFSKeyConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorContract;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.kfs.vnd.businessobject.VendorRoutingComparable;
+import org.kuali.kfs.vnd.document.service.VendorService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kns.util.KNSGlobalVariables;
+import org.kuali.rice.kns.util.MessageList;
+import org.kuali.rice.krad.bo.Note;
+import org.kuali.rice.krad.bo.PersistableBusinessObjectExtension;
+import org.kuali.rice.krad.document.Document;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CuDisbursementVoucherDocumentTest {
+
+    @Mock
+    private static CuDisbursementVoucherDocument cuDisbursementVoucherDocument;
+    private static VendorService vendorService;
+    private static DisbursementVoucherPayeeService disbursementVoucherPayeeService;
+    private static DisbursementVoucherPaymentReasonService disbursementVoucherPaymentReasonService;
+    private static CuDisbursementVoucherPayeeDetail dvPayeeDetail;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        ArrayList<String> methodNames = new ArrayList<>();
+        for (Method method : CuDisbursementVoucherDocument.class.getMethods()) {
+            if (!Modifier.isFinal(method.getModifiers()) && !method.getName().startsWith("set") && !method.getName().startsWith("get")) {
```

Comment:
Yeah, I want to review this with the team sometime. It will be easier to discuss rather than just comment on here.